src/DemoFile/DemoParser.FullPacket.cs

```diff
@@ -0,0 +1,161 @@
+using System.Diagnostics;
+
+namespace DemoFile;
+
+public partial class DemoParser
+{
+    private readonly record struct FullPacketPosition(DemoTick Tick, long StreamPosition)
+        : IComparable<FullPacketPosition>
+    {
+        public int CompareTo(FullPacketPosition other) => Tick.CompareTo(other.Tick);
+    }
+
+    /// Key ticks occur every 60 seconds
+    private const int FullPacketInterval = 64 * 60;
+
+    // todo: reset on StartReadingAsync
+    private readonly List<FullPacketPosition> _fullPacketPositions = new(64);
+    private DemoTick _readFullPacketTick = DemoTick.PreRecord;
+    private int _fullPacketTickOffset;
+
+    private bool TryFindFullPacketBefore(DemoTick demoTick, out FullPacketPosition fullPacket)
+    {
+        var idx = _fullPacketPositions.BinarySearch(new FullPacketPosition(demoTick, 0L));
+        if (idx >= 0)
+        {
+            fullPacket = _fullPacketPositions[idx];
+            return true;
+        }
+
+        var fullPacketBeforeIdx = ~idx - 1;
+        if (fullPacketBeforeIdx >= 0 && fullPacketBeforeIdx < _fullPacketPositions.Count)
+        {
+            fullPacket = _fullPacketPositions[fullPacketBeforeIdx];
+            return true;
+        }
+
+        fullPacket = default;
+        return false;
+    }
+
+    public async ValueTask SeekToTickAsync(DemoTick tick, CancellationToken cancellationToken)
+    {
+        // todo: throw if currently in a MoveNextAsync
```

Comment:
Disable seeking inside of command parsing

Reply:
Maybe also prevent seek while already seeking ? Expose a property `IsSeeking` ?